docs/troubleshooting.asciidoc

```diff
@@ -18,6 +18,17 @@ IMPORTANT: If you do so, *please attach your debug logs* so that we can analyze
 Upload the *complete* logs to a service like https://gist.github.com.
 The logs should include everything from the application startup up until the first request has been executed.
 
+[float]
+[[trouble-shooting-additional-agent]]
+=== Running alongside other agents
+Like many other Java agents, our agent instruments classes by injecting bytecode into them on runtime. Our bytecode
+instrumentation guarantees to produce only valid bytecode and to never change the class's "schema". If you are using
+other Java agents in addition to ours that have the same guarantees, they should not interfere with each other.
+However, some Java agents do not conform to these guarantees, in which case there may be issues with using them in
+parallel to ours. If you encounter errors, one of the first things to do is to remove any additional agent in order to
+check whether this is indeed a case of agent mismatch. We may still find a way to make them work together, but this
+information would be cruicial for our ability to assist, so make sure to include it when reporting such issues.
```

Comment:
:heart_eyes: I love being able to link to this paragraph whenever there are multiple agents. Should we do the same for `-Xverify:none` too ?

Reply:
We decided to be more blunt and restrictive with disabled BC verification - it will be handled through https://github.com/elastic/apm-agent-java/pull/1593. The current plan is to issue a warning when we encounter that in a few versions later disable ourselves when it is used, enforcing the user to disable the bootstrap checks.